packages/eslint-plugin/src/rules/consistent-type-definitions.ts

```diff
@@ -122,6 +122,19 @@ export default util.createRule({
                     });
                   }
 
+                  if (
```

Comment:
While working on this fix, I noticed that the option is checked **after** having already found a certain node (and in some other files as well), for example:

https://github.com/typescript-eslint/typescript-eslint/blob/1e7e6d8cec0bae3736ae2c8a84509b7dddc2525f/packages/eslint-plugin/src/rules/consistent-type-definitions.ts#L50-L54

<details>
<summary>Other cases <sub><i>(click to expand)</i></sub></summary>

https://github.com/typescript-eslint/typescript-eslint/blob/b1d44497ddf8ef4615c727fe4241e54f7f1aba65/packages/eslint-plugin/src/rules/method-signature-style.ts#L120-L123

https://github.com/typescript-eslint/typescript-eslint/blob/1c1b572c3000d72cfe665b7afbada0ec415e7855/packages/eslint-plugin/src/rules/restrict-plus-operands.ts#L136-L137

https://github.com/typescript-eslint/typescript-eslint/blob/25ea953cc60b118bd385c71e0a9b61c286c26fcf/packages/eslint-plugin/src/rules/sort-type-union-intersection-members.ts#L242-L244

https://github.com/typescript-eslint/typescript-eslint/blob/1c1b572c3000d72cfe665b7afbada0ec415e7855/packages/eslint-plugin/src/rules/typedef.ts#L180-L183

https://github.com/typescript-eslint/typescript-eslint/blob/1c1b572c3000d72cfe665b7afbada0ec415e7855/packages/eslint-plugin/src/rules/explicit-function-return-type.ts#L71-L76

</details>

But, IMHO this could be improved by moving the conditional up then wrapping the `return {`, and so avoiding looking for nodes which doesn't corresponds unnecessarily. Does it makes sense for you? If so, I can send a PR `refactor` for that.

Reply:
happy to accept a PR if you want!